roofit/roofitcore/src/TestStatistics/RooSumL.cxx

```diff
@@ -0,0 +1,90 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+#include <TestStatistics/RooSumL.h>
+#include <RooAbsData.h>
+#include <TestStatistics/RooSubsidiaryL.h>
+
+#include <algorithm> // min, max
+
+namespace RooFit {
+namespace TestStatistics {
+
+// Note: components must be passed with std::move, otherwise it cannot be moved into the RooSumL because of the unique_ptr!
+RooSumL::RooSumL(RooAbsPdf *pdf, RooAbsData *data, std::vector<std::unique_ptr<RooAbsL>> components, RooAbsL::Extended extended)
+   : RooAbsL(pdf, data,
+             data->numEntries(), // TODO: this may be misleading, because components in reality will have their own N_events...
+             components.size(), extended), components_(std::move(components))
+{}
+
+
+bool RooSumL::processEmptyDataSets() const
+{
+   // TODO: check whether this is correct! This is copied the implementation of the RooNLLVar override; the
+   // implementation in RooAbsTestStatistic always returns true
+   return extended_;
+}
+
+double RooSumL::evaluatePartition(Section events, std::size_t components_begin, std::size_t components_end)
+{
+   // Evaluate specified range of owned GOF objects
+   double ret = 0;
+
+   // from RooAbsOptTestStatistic::combinedValue (which is virtual, so could be different for non-RooNLLVar!):
+   eval_carry_ = 0;
+   for (std::size_t ix = components_begin; ix < components_end; ++ix) {
+      // TODO: make sure we only calculate over events in the sub-range that the caller asked for
+      //      std::size_t component_events_begin = std::max(events_begin, components_[ix]->get_N_events())  // THIS
+      //      WON'T WORK, we need to somehow allow evaluate_partition to take in separate event ranges for all
+      //      components...
```

Comment:
Remove / Fix / test / ...

We don't commented out code, and TODOs only in exceptional circumstances.

Reply:
This was fixed.